hamilton/include/hamilton/hybrid_planner.h

```diff
@@ -0,0 +1,195 @@
+#include <descartes_trajectory/joint_trajectory_pt.h>
+#include <descartes_planner/dense_planner.h>
+#include <descartes_planner/sparse_planner.h>
+#include <visualization_msgs/MarkerArray.h>
+#include <eigen_conversions/eigen_msg.h>
+// #include <ur5_demo_descartes/ur5_robot_model.h>
+#include <pluginlib/class_loader.h>
+#include <descartes_moveit/moveit_state_adapter.h>
+#include <geometry_msgs/PoseArray.h>
+//TODO instead of specifying traj, planner and their subtypes explicitly -> find a simpler way
+
+
+//TODO namespace anon vs named
+namespace hamilton
+{
+//TODO TOCHECK ros params vs protected variables (vs partially hard coded)
+const std::string ROBOT_DESCRIPTION_PARAM = "robot_description";
+const std::string EXECUTE_TRAJECTORY_SERVICE = "execute_kinematic_path";
+const std::string VISUALIZE_TRAJECTORY_TOPIC = "visualize_trajectory_curve";
+const double SERVICE_TIMEOUT = 5.0f; // seconds
+const double ORIENTATION_INCREMENT = 0.5f;
+const double EPSILON = 0.0001f;
+const double AXIS_LINE_LENGHT = 0.01;
+const double AXIS_LINE_WIDTH = 0.001;
+const std::string PLANNER_ID = "RRTConnectkConfigDefault";
+const std::string HOME_POSITION_NAME = "home";
+//TOCHECK config_ members or hardcoded MoveIt! params
+
+typedef std::vector<descartes_core::TrajectoryPtPtr> descartes_traj;
+typedef descartes_trajectory::CartTrajectoryPt descartes_cart_pt;
+// typedef descartes_trajectory::CartTrajectoryPt::AxialSymmetricPt descartes_cart_ax_sym_pt;
+typedef descartes_trajectory::JointTrajectoryPt descartes_joint_traj_pt;
+typedef moveit_msgs::RobotTrajectory moveit_robot_traj;
+typedef trajectory_msgs::JointTrajectory moveit_joint_traj;
+typedef boost::shared_ptr<move_group_interface::MoveGroup> MoveGroupPtr;
+// typedef moveit_msgs::RobotTrajectory::joint_trajectory moveit_joint_traj;
+// typedef trajectory_msgs::JointTrajectory moveit_joint_traj;
+// typedef descartes_core::RobotModelPtr descartes_robot_model_ptr;
+
+
+
+/*  =============================== Application Data Structure ===============================
+ *
+ * Holds the data used at various points in the application.  This structure is populated
+ * from data found in the ros parameter server at runtime.
+ *
+ */
+
+ //TODO TOCHECK HybridConfiguration should be broken into three structs:
+ // - MoveIt! params
+ // - Descartes params
+ // - Common or global params
+
+ //Params should be entered from GUI. Adapt Fermi for this?
+
+struct HybridConfiguration
+{
+  std::string group_name;                 /* Name of the manipulation group containing the relevant links in the robot */
+  std::string tip_link;                   /* Usually the last link in the kinematic chain of the robot */
+  std::string base_link;                  /* The name of the base link of the robot */
+  std::string world_frame;                /* The name of the world link in the URDF file */
+  std::vector<std::string> joint_names;   /* A list with the names of the mobile joints in the robot */
+
```

Comment:
I'd recommend to break them down into categories if you end up managing too many parameters.  Also, you could namespace the parameters in order to keep things more organized.  However approach you take just remember to document it.